ipn/ipnlocal/local.go

```diff
@@ -2971,6 +2975,14 @@ func (b *LocalBackend) applyPrefsToHostinfo(hi *tailcfg.Hostinfo, prefs ipn.Pref
 		sshHostKeys = b.getSSHHostKeyPublicStrings()
 	}
 	hi.SSH_HostKeys = sshHostKeys
+
+	// The Hostinfo.WantIngress field tells control whether this node wants to
+	// be wired up for ingress connections. If harmless if it's accidentally
+	// true; the actual policy is controlled in tailscaled by ServeConfig. But
+	// if this is accidentally false, then control may not configure DNS
+	// properly. This exists as an optimization to control to program fewer DNS
+	// records that have ingress enabled but are not actually being used.
+	hi.WireIngress = b.wantIngressLocked() || prefs.WireIngress()
```

Comment:
Why is this not &&. Having two ways of controlling the same bit sounds scary

Reply:
It's a hedge against the logic being wrong. The pref should be right and synced with wantIngressLocked (serveConfig), though. There's no downside to accidentally sending true. It's just an optimization for control.

There is, however, a downside to accidentally sending the wrong value: it doesn't get wired up by control.
 
So I'm erring on sending true.

Currently we send nothing and effectively treat all nodes as sending true.

Reply:
PTAL. Redid it all.